File: src/main/java/com/mbayou/kick4k/authorization/OAuthTokenResponse.java
package com.mbayou.kick4k.authorization;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class OAuthTokenResponse {
    private final String accessToken;
    private final String tokenType;
    private final String refreshToken;
    private final Integer expiresIn;
    private final String scope;

    @JsonCreator
    public OAuthTokenResponse(@JsonProperty("access_token") String accessToken,
                              @JsonProperty("token_type") String tokenType,
                              @JsonProperty("refresh_token") String refreshToken,
                              @JsonProperty("expires_in") Integer expiresIn,
                              @JsonProperty("scope") String scope) {
        this.accessToken = accessToken;
        this.tokenType = tokenType;
        this.refreshToken = refreshToken;
        this.expiresIn = expiresIn;
        this.scope = scope;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public String getTokenType() {
        return tokenType;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public Integer getExpiresIn() {
        return expiresIn;
    }

    public String getScope() {
        return scope;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/authorization/OAuthTokenResponse.kt
package com.mbayou.kick4k.authorization

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class OAuthTokenResponse @JsonCreator constructor(
    @JsonProperty("access_token") val accessToken: String,
    @JsonProperty("token_type") val tokenType: String,
    @JsonProperty("refresh_token") val refreshToken: String,
    @JsonProperty("expires_in") val expiresIn: Int,
    @JsonProperty("scope") val scope: String,
    @JsonProperty("refresh_expires_in") val refreshExpiresIn: Int?,
) {
    constructor(
        accessToken: String,
        tokenType: String,
        refreshToken: String,
        expiresIn: Int,
        scope: String,
    ) : this(accessToken, tokenType, refreshToken, expiresIn, scope, null)
}
